### PR TITLE
Generalized file paths for components (app)

### LINKED ISSUE #68: Move template.hbs alongside route.js or component.js
Ember's component blueprint will nest components under the `app/components/` directory. However it's also valid Ember to place components directly in a pod folder under the `app/` root.

Both of these are valid components
```
app/
  components/
    my-pod-b/
      component.js
      template.hbs
  my-pod-a/
    component.js
    template.hbs
```

Problem is route templates pod similarly to `my-pod-a`, and if all you're looking at is template.hbs you have no way to know if it's a route or a component template. You need the broader context of there being a route.js or component.js present.

If you have components stored like `my-pod-a` and run this codemod, the map-route-template will move the component's template.hbs into the `app/templates/` folder, and map-component-classes will never move the component.js. This breaks the component.

I don't see a way to fix this with the current architecture. I think there needs to be a more broad map-route and map-component concept which, upon spotting a route.js or component.js, also look for the sibling template.hbs and move it appropriately.

### PR DESCRIPTION
**Pull request by ijlee2**
## Background

Closes #68.

For apps, the codemod can now find component files that were placed outside of the usual `/components` folder. To be considered, there must be a class file named `component.d.ts`, `component.js`, or `component.ts`.

### CODE REVIEW

File: tests/migration/app/steps/create-file-path-maps/edge-case-components.test.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+          'app/components/unknowns/example-1.js',
+        ],
+        [
+          'app/unknowns/example-2/styles.css',
+          'app/styles/unknowns/example-2.css',
+        ],
+        [
+          'app/unknowns/example-3/template.hbs',
+          'app/templates/unknowns/example-3.hbs',
+        ],
```
Comment (ijlee2) [inline tests/migration/app/steps/create-file-path-maps/edge-case-components.test.ts:155]: Edge cases (single file).
